File: 고득점 kit/타겟 넘버.swift
import Foundation

var answer:Int = 0
var visited:[Bool] = Array(repeating: false, count: 20)

func DFS(array: [Int], aim: Int, tot: Int, level: Int) {
    // 도달했다면
    if level == array.count {
        if aim == tot { // 같다면
            answer += 1
        }
        return
    }
    
    // 더하기 빼기 둘다 진행
    DFS(array: array, aim: aim, tot: tot + array[level], level: level + 1)
    DFS(array: array, aim: aim, tot: tot - array[level], level: level + 1)
    return
}

func solution(_ numbers:[Int], _ target:Int) -> Int {
    // DFS 이용
    DFS(array: numbers, aim: target, tot:0, level: 0)
    return answer
}

File: self-taught graffiti/고득점 kit/DFSBFS/타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

void DFS(vector<int>& numbers, int target, int tot, int level, int &answer) {
    if (level == numbers.size()) {
        if (tot == target)
            answer++;
        return;
    }
    
    DFS(numbers, target, tot + numbers[level], level + 1, answer);
    DFS(numbers, target, tot - numbers[level], level + 1, answer);
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    
    DFS(numbers, target, 0, 0, answer);
    return answer;
}